src/Shared/HttpSys/NativeInterop/HttpApiTypes.cs

```diff
@@ -599,10 +599,10 @@ internal struct HTTP_SERVER_AUTHENTICATION_INFO
     {
         internal HTTP_FLAGS Flags;
         internal HTTP_AUTH_TYPES AuthSchemes;
-        internal bool ReceiveMutualAuth;
-        internal bool ReceiveContextHandle;
-        internal bool DisableNTLMCredentialCaching;
-        internal ulong ExFlags;
+        internal byte ReceiveMutualAuth;
```

Comment:
I'd prefer to retain at least a comment indicating that it's boolean.

Reply:
> I'd prefer to retain at least a comment indicating that it's boolean.

With 7f18f8fea5c8e2efc26050f0815f8c911bb26ff1, this change seems to be no longer necessary for `main`.

However, since c50346d52569c318f6a198f03d9e0a517cda8abe seems to fix a rather significant issue with an invalid structure layout/size, is it something that should be backported to stable branches?

Please let me know if that is the case, and whether you'd want me to handle that with a separate PR or in any other convenient way.

Reply:
Can you demonstrate any negative effects from the old structure? We've not had any complaints here.

Reply:
> Can you demonstrate any negative effects from the old structure? We've not had any complaints here.

In my x64 environment, the native size of the structure is:
```
sizeof(HTTP_SERVER_AUTHENTICATION_INFO) == 64
```

In the same environment, the size of the interop structure before the fix is:
```
Marshal.SizeOf<HttpApiTypes.HTTP_SERVER_AUTHENTICATION_INFO> == 80
```

In turn, this means that:
1. The field layout is wrong after the `AuthSchemes` field
2. An invalid (larger) size is passed during the call to `HttpSetUrlGroupProperty()`

Maybe 1. hasn't been causing any practical issues because the current version of the code didn't change any structure fields after the `AuthSchemes`. But I would still say that 2. is quite significant just by itself: the behavior when passing an invalid size to the `HttpSetUrlGroupProperty()` function seems to be unspecified, and could have easily resulted in an error or even a crash.

In my environment the call seems to succeed, so from that perspective this remains a theoretical issue, but, to be precise, there is a chance that this behavior may change in the future or may differ depending on the OS version — I haven't checked that in detail.

-----

For reference, here is the native definition of the related typedefs and the structure itself:
```
typedef BYTE  BOOLEAN;
typedef unsigned char UCHAR;

typedef struct _HTTP_SERVER_AUTHENTICATION_INFO
{
    HTTP_PROPERTY_FLAGS Flags;

    ULONG AuthSchemes;

    BOOLEAN ReceiveMutualAuth;
    BOOLEAN ReceiveContextHandle;
    BOOLEAN DisableNTLMCredentialCaching;

    UCHAR   ExFlags;

    HTTP_SERVER_AUTHENTICATION_DIGEST_PARAMS DigestParams;
    HTTP_SERVER_AUTHENTICATION_BASIC_PARAMS  BasicParams;

} HTTP_SERVER_AUTHENTICATION_INFO, *PHTTP_SERVER_AUTHENTICATION_INFO;
```

Reply:
Ok, we won't consider backporting it without a repro of a functional issue. For now it's enough that the issue is gone in main.